integration/test/ParseObjectTest.js

```diff
@@ -304,10 +304,7 @@ describe('Parse Object', () => {
     });
 
     obj.increment('objectField.number', 15);
```

Comment:
The following test fails. Can you add it to the PR?

```
it('can increment array nested fields', async () => {
    const obj = new TestObject();
    obj.set('items', [ { value: 'a', count: 5 }, { value: 'b', count: 1 } ]);
    await obj.save();

    obj.increment('items.0.count', 15);
    await obj.save();

    const query = new Parse.Query(TestObject);
    const result = await query.get(obj.id);
    assert.equal(result.get('items')[0].count, 20);
  });
```

Reply:
How would we differentiate between:
```
{
  "a": {
    "b": [
      { "c": 0 }
    ]
  }
}
```
and
```
{
  "a": {
    "b": {
      "0": {
        "c": 0
      }
    }
  }
}
```

Reply:
@mtrezza https://github.com/parse-community/Parse-SDK-JS/pull/1451 fixes handles applying logic internally. The only problem is the Server. The server treats all dot notation as type Object so using dot notation an Array will result in a schema mismatch.